src/openassetio-core/include/openassetio/hostApi/Manager.hpp

```diff
@@ -101,6 +101,128 @@ class OPENASSETIO_CORE_EXPORT Manager final {
    */
   [[nodiscard]] Str displayName() const;
 
+  /**
+   * @}
+   */
+
+  /**
+   * @name Asset Management System Information
+   *
+   * These functions provide general information about the @ref
+   * asset_management_system itself.
+   * @{
+   */
+
+  /**
+   * Capabilities that the manager implements.
+   *
+   * Many OpenAssetIO methods are optional. This enum is used with the
+   * introspection mechanism @ref hasCapability to provide a means of
+   * querying which sets of methods the manager provides.
+   */
+  enum class Capability : int {
+    /**
+     * Manager makes use of the context to persist custom state for
+     * performance reasons or otherwise.
+     *
+     * @note If this capability is true, then the host must reuse the
+     * same context across related API calls (including the use of @ref
+     * persistenceTokenForContext when the calls are distributed
+     * cross-process).
+     *
+     * @see @ref stable_resolution
+     */
+    kStatefulContexts =
+        static_cast<int>(managerApi::ManagerInterface::Capability::kStatefulContexts),
```

Comment:
[question] It still bothers me that we got rid of the forward declaration of `ManagerInterface` in order to support this conforming of enum values.

Alternatives:

* The pattern established for `Access`, where independent `enum class`es  have a common `enum` dependency.  But that requires additional boilerplate (i.e. an extra public header).
* IIRC the key reason to conform the values was for indexing into `kCapabilityNames`, so another solution would be to duplicate and customise that. Not very DRY, runs the risk of accidentally indexing into the wrong `kCapabilityNames` , and makes the forwarding of enum value  from `Manager` to `ManagerInterface` more complex.

Reply:
For me, the main reason was for the avoidance of confusion ie making it very clear that `Manager::kPublishing` _is and always must be_ the same as `ManagerInterface::kPublishing`. We couldn't think of any logical reason to diverge the manager constants from interface constants. The only real scenario was the one we have where the Manager ones are a sub-set of Interface ones. 

We did talk about the potential need to add new ones to `Manager` as a "collection of capabilities", but then realized that this would most likely be done via some other mechanism to a) include `managementPolicy` and b) ensure the capabilities in the set were discoverable (which they wouldn't be if we added them here).

 I'm not to worried about how it is right now as to me, the `ManagerInterface` enum _is_ the base enum, and unlike access, there is only _one_ case these are queried (access has several logically different operations that share similar concepts). 

Is the removal of the forward declare a deal breaker?

Reply:
Sorry, forgot - if we factor it into a base enum (and this is probably true for access), it its potentially more confusing than the access one for consumers, as we don't have a lot of scope to make the names more meaningful for the "enums you use"? And we don't have a `private` public header space yet do we?

Reply:
Thinking more generally: we could perhaps do with a consistent policy re. enums, especially where shared between `Manager` and `ManagerInterface`.  E.g. I'm wondering if `Access` enums should be duplicated by `Manager` and `ManagerInterface`, allowing us to doc them independently, and decouple them so they can to vary independently (though we don't have a use case for that, it seems good practice).

Perhaps having a shared header with a root enum, which `Manager` and `ManagerInterface` then duplicate a subset from, would be a good consistent universal pattern?

Reply:
[Capturing some of offline conversation] It turns out we don't (yet) have a general problem of consistency between enums, there really is only `Access` and `Capability` that are manager-specific, and they are both quite different use-cases.  So I'm less concerned with consistency now.

I'm still wary of losing the forward-declaration of `ManagerInterface.hpp`, but I do wonder if avoiding it this time is just postponing the inevitable...

Reply:
After more offline discussion, added c6ed7dbf, which creates a new `internal` namespace with both the `Access` and `Capability` root enum values.  How is that?